libpcsxcore/psxinterpreter.c

```diff
@@ -176,25 +176,21 @@ static int execBreakCheck(psxRegisters *regs, u32 pc)
 // get an opcode without triggering exceptions or affecting cache
 u32 intFakeFetch(u32 pc)
 {
-	u8 *base = psxMemRLUT[pc >> 16];
-	u32 *code;
-	if (unlikely(base == INVALID_PTR))
+	u32 *code = (u32 *)psxm(pc & ~0x3, 0);
+	if (unlikely(code == INVALID_PTR))
 		return 0; // nop
-	code = (u32 *)(base + (pc & 0xfffc));
 	return SWAP32(*code);
 
 }
 
-static u32 INT_ATTR fetchNoCache(psxRegisters *regs, u8 **memRLUT, u32 pc)
```

Comment:
Could you keep the memRLUT pointer? Since it's such a hot spot in the interpreter, I remember that reloading this pointer from the PIC area for each fetch being a perf hit, that's why it's passed around. You can then have something like
```
static inline void * psxm_(u32 mem, int write, u8 **memRLUT)
{
   ...
}
static inline void * psxm(u32 mem, int write)
{
    return psxm_(mem, write, psxMemRLUT);
}
```

Reply:
Sure. Although this could be said for a lot of the code, IMHO there are way too many global variables that could be in a state structure instead, that would be passed around to functions as a pointer (kind of doing object-oriented C).